diff -r -U4 cpufetch-1.03--orig/src/common/printer.c cpufetch-1.03--patched/src/common/printer.c
--- cpufetch-1.03--orig/src/common/printer.c	2023-01-05 11:17:49.000000000 +0100
+++ cpufetch-1.03--patched/src/common/printer.c	2023-01-05 12:36:03.274338380 +0100
@@ -483,13 +483,16 @@
 
       if(attr_type == ATTRIBUTE_L3) {
         add_space = false;
       }
+#if defined(ARCH_X86) || defined(ARCH_ARM)
       if(attr_type == ATTRIBUTE_CPU_NUM) {
         printOut(lbuf, strlen(attr_value), "%s%s%s", logo->color_text[0], attr_value, art->reset);
         add_space = true;
       }
-      else {
+      else
+#endif
+      {
         beg_space = 0;
         space_right = 2 + 1 + (la - strlen(attribute_fields[attr_type]));
         if(hybrid_architecture && add_space) {
           beg_space = 2;
@@ -689,9 +692,9 @@
     attribute_fields = ATTRIBUTE_FIELDS_SHORT;
     longest_attribute = longest_attribute_length(art, attribute_fields);
   }
 
-  print_ascii_generic(art, longest_attribute, term->w, attribute_fields);
+  print_ascii_generic(art, longest_attribute, term->w, attribute_fields, false);
 
   return true;
 }
 #endif
